mono/sgen/sgen-pinning-stats.c

```diff
@@ -220,7 +220,7 @@ sgen_pin_stats_report (void)
 		int i;
 		mono_gc_printf (sgen_gc_debug_file, "%-50s", name);
 		for (i = 0; i < PIN_TYPE_MAX; ++i)
-			mono_gc_printf (sgen_gc_debug_file, "  %10ld", pinned_entry->num_pins [i]);
+			mono_gc_printf (sgen_gc_debug_file, "  %10zd", pinned_entry->num_pins [i]);
```

Comment:
Format already used in several places around the source code and should be an addition to C99 and mandatory in C11, that's why I picked it. I believe the only compiler that still struggles is mingw at this point, but it already has problems with a bunch of other formats in various files, so fixing mingw will need to make an overhaul at several places in source code. Having that said I agree with more portable way and since it already limited to %ld in current code cast to long should give same result without warning on VS2015.